benchmarks/benchmark_shared.c

```diff
@@ -142,19 +147,24 @@ int bandwidth(void)
 				ft_start();
 
 			for(j = 0; j < opts.window_size; j++) {
-				if (opts.transfer_size < fi->tx_attr->inject_size)
+				if (opts.transfer_size < fi->tx_attr->inject_size) {
 					ret = ft_inject(opts.transfer_size);
-				else
-					ret = ft_post_tx(opts.transfer_size);
+					if (ret == -FI_EAGAIN)
+						ret = ft_post_tx(
+							opts.transfer_size,
```

Comment:
If ft_inject returns EAGAIN, then there is no guarantee that ft_post_tx also won't return EAGAIN.

In that sense, we should have a better handling of EAGAIN that is consistent. My thought, is that we should remove this branch for now and handle EAGAIN better inside the ft_inject or ft_post_tx calls.